core/dbt/contracts/publication.py

```diff
@@ -0,0 +1,110 @@
+from typing import Optional, List, Dict, Any
+from datetime import datetime
+from dbt.dataclass_schema import dbtClassMixin
+
+from dataclasses import dataclass, field
+
+from dbt.contracts.util import BaseArtifactMetadata, ArtifactMixin, schema_version
+from dbt.contracts.graph.unparsed import NodeVersion
+from dbt.contracts.graph.nodes import ManifestOrPublicNode
+from dbt.node_types import NodeType, AccessType
+
+
+@dataclass
+class ProjectDependency(dbtClassMixin):
+    name: str
+    environment: str
```

Comment:
We can get rid of this field for now as we don't have plans to use it in core for 1.6. That said, allowing for additional attributes in `ProjectDependency` would enable projects to set up additional configuration that could be useful to them in fetching publication artifacts.

Reply:
Do we want to allow random keys here? We could do that by including an additional base class that puts them in an _extra dictionary.

Reply:
Yep, I think that'd be appropriate here.

Reply:
Let's make that a follow-on ticket.